ke/compiler.py

```diff
@@ -89,6 +89,47 @@
     builtin_macros["#" + short] = builtin_macros["#" + long]
 
 
+def compile_separate(separate, expr):
+    print('\n\n\n\n#########')
+
+    if separate is None:
+        raise CompileError("Must specify a valid separator: ',', ':', '|'")
+    print(f"incoming separator of: '{separate}' - '{expr}'")
+    (a, b, is_greedy, sub) = expr
+    if sub is None:
+        return CompileError("Must specify a token")
+
+    if b is not None and a > b:
+        return CompileError("Min can not exceed Max.")
```

Comment:
how about you move this check to where it would benefit all ranges and not just separated ones? but also improve the message, e.g. "Range upper bound exceeds lower bound"

Reply:
What do you mean all ranges? They special case about this one is where b might be None, which is acceptable, and might not be in other checks?